indico/modules/events/api.py

```diff
@@ -538,6 +512,31 @@ def _update_query(self, query):
     def serialize_events(self, events):
         return map(self._build_event_api_data, events)
 
+    def _serialize_category_path(self, id):
+        category = Category.get(id)
```

Comment:
You use this method only in one place, so you can simply have one that queries all needed categories at once (and undefers `chain`) and then serializes them in a loop